### PR TITLE
Solution is ready

### PR DESCRIPTION
**Pull request by Likh-Alex**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,13 +3,37 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE_OF_ARRAY = 10;
+    private final Object[] keys = new Object[SIZE_OF_ARRAY];
+    private final Object[] values = new Object[SIZE_OF_ARRAY];
+    private int currentIndex = 0;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]: let's initialise all fields in the constructor

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,13 +3,37 @@
...
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < SIZE_OF_ARRAY; i++) {
+            if (keys[i] == key) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:14]: is it the right way to compare two objects?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,13 +3,37 @@
...
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE_OF_ARRAY = 10;
+    private final Object[] keys = new Object[SIZE_OF_ARRAY];
+    private final Object[] values = new Object[SIZE_OF_ARRAY];
+    private int currentIndex = 0;
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < SIZE_OF_ARRAY; i++) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:13]: should we iterate from 0 to 9 here or can we optimise this?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,13 +3,37 @@
...
     @Override
     public V get(K key) {
+        if (key == null) {
+            for (int j = 0; j < SIZE_OF_ARRAY; j++) {
+                if (keys[j] == null) {
+                    return (V) values[j];
+                }
+            }
+        }
+        for (int i = 0; i < SIZE_OF_ARRAY; i++) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:32]: same here

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,13 +3,37 @@
...
+            if (keys[i] != null && keys[i].equals(key)) {
+                return (V) values[i];
+            }
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:36]: can we write all these checks in a single loop and in a single`if`?
remember how we compared two objects in previous tasks (we compared fields there)

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: done

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: changed comparison to equals()

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: refactored iteration up to "current index" only

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 4 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: refactored iteration up to "current index" only here also

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 5 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: tried to fit all checks in one iteration and one if statement

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,13 +3,38 @@
...
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < currentIndex; i++) {
+            if (keys[i] == key || keys[i] != null && keys[i].equals(key)) {
+                values[i] = value;
+            }
+        }
+        keys[currentIndex] = key;
+        values[currentIndex] = value;
+        currentIndex++;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:26]: these  lines are currently executed in any case, no matter if you replaced the value earlier

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -3,13 +3,38 @@
...
     }
 
     @Override
     public V get(K key) {
+        for (int j = 0; j < currentIndex; j++) {
+            if (key == null & keys[j] == null) {
+                return (V) values[j];
+            } else if (key != null && key.equals(keys[j])) {
+                return (V) values[j];
+            }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:36]: look at the `put()` method and write everything in one `if` **here**

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 6 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:27]: add return statement after overwriting an object

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 7 above)
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: all checks are in one `if` statement now

**Review by KseniiaMakarova — APPROVED** [review]
great job!

**Pull request closed without merging**